Skype/Trusted-Application-API/samples/QuickStartSamples/InviteAndRemoveParticipant/Program.cs

```diff
@@ -0,0 +1,153 @@
+            var applicationName = ConfigurationManager.AppSettings["Trouter_ApplicationName"];
+            var userAgent = ConfigurationManager.AppSettings["Trouter_UserAgent"];
+            var participantUri = ConfigurationManager.AppSettings["ParticipantUri"];
+            var token = SkypeTokenClient.ConstructSkypeToken(
+                skypeId: skypeId,
+                password: password,
+                useTestEnvironment: false,
+                scope: string.Empty,
+                applicationName: applicationName).Result;
+
+            m_logger = new ConsoleLogger();
+
+            // Uncomment for debugging
+            // m_logger.HttpRequestResponseNeedsToBeLogged = true;
+
+            EventChannel = new TrouterBasedEventChannel(m_logger, token, userAgent);
+
+            // Prepare platform
+            var platformSettings = new ClientPlatformSettings(QuickSamplesConfig.AAD_ClientSecret, new Guid(QuickSamplesConfig.AAD_ClientId));
+            var platform = new ClientPlatform(platformSettings, m_logger);
+
+            // Prepare endpoint
+            var endpointSettings = new ApplicationEndpointSettings(new SipUri(QuickSamplesConfig.ApplicationEndpointId));
+            var applicationEndpoint = new ApplicationEndpoint(platform, endpointSettings, EventChannel);
+
+            var loggingContext = new LoggingContext(Guid.NewGuid());
+            await applicationEndpoint.InitializeAsync(loggingContext).ConfigureAwait(false);
+            await applicationEndpoint.InitializeApplicationAsync(loggingContext).ConfigureAwait(false);
+
+            // Meeting configuration
+            var meetingConfiguration = new AdhocMeetingCreationInput(Guid.NewGuid().ToString("N") + " test meeting");
+
+            // Schedule meeting
+            var adhocMeeting = await applicationEndpoint.Application.CreateAdhocMeetingAsync(loggingContext, meetingConfiguration).ConfigureAwait(false);
+
+            WriteToConsoleInColor("ad hoc meeting uri : " + adhocMeeting.OnlineMeetingUri);
+            WriteToConsoleInColor("ad hoc meeting join url : " + adhocMeeting.JoinUrl);
+
+            // Get all the events related to join meeting through Trouter's uri
+            platformSettings.SetCustomizedCallbackurl(new Uri(EventChannel.CallbackUri));
+
+            // Start joining the meeting
+            var invitation = await adhocMeeting.JoinAdhocMeeting(loggingContext, null).ConfigureAwait(false);
+
+            // Wait for the join to complete
+            await invitation.WaitForInviteCompleteAsync().ConfigureAwait(false);
+
+            invitation.RelatedConversation.HandleParticipantChange += Conversation_HandleParticipantChange;
+
+            // invite the participant to join the meeting
+            WriteToConsoleInColor("Invite " + participantUri+" to join the meeting");
+            await invitation.RelatedConversation.AddParticipantAsync(participantUri, loggingContext).ConfigureAwait(false);
```

Comment:
This starts an IParticipantInvitation. You can use WaitForInviteCompleteAsync() on it to wait for the participant to accept it and then eject the user.